codesign.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/retry"
+	"github.com/bitrise-io/go-xcode/autocodesign/certdownloader"
+	"github.com/bitrise-io/go-xcode/autocodesign/codesignasset"
+	"github.com/bitrise-io/go-xcode/autocodesign/devportalclient"
+	"github.com/bitrise-io/go-xcode/autocodesign/localcodesignasset"
+	"github.com/bitrise-io/go-xcode/autocodesign/projectmanager"
+	"github.com/bitrise-io/go-xcode/codesign"
+	"github.com/bitrise-io/go-xcode/devportalservice"
+)
+
+func createCodesignManager(config Config, xcodeMajorVersion int64, logger log.Logger, cmdFactory command.Factory) (codesign.Manager, error) {
+	var authType codesign.AuthType
+	switch config.CodeSigningAuthSource {
+	case codeSignSourceAppleID:
+		authType = codesign.AppleIDAuth
+	case codeSignSourceAPIKey:
+		authType = codesign.APIKeyAuth
+	case codeSignSourceOff:
+		return codesign.Manager{}, fmt.Errorf("automatic code signing is disabled")
+	}
+
+	codesignInputs := codesign.Input{
+		AuthType:                  authType,
+		DistributionMethod:        "development",
+		CertificateURLList:        config.CertificateURLList,
+		CertificatePassphraseList: config.CertificatePassphraseList,
+		KeychainPath:              config.KeychainPath,
+		KeychainPassword:          config.KeychainPassword,
+	}
+
+	codesignConfig, err := codesign.ParseConfig(codesignInputs, cmdFactory)
+	if err != nil {
+		return codesign.Manager{}, fmt.Errorf("issue with input: %s", err)
+	}
+
+	var serviceConnection *devportalservice.AppleDeveloperConnection = nil
+	devPortalClientFactory := devportalclient.NewFactory(logger)
+	if authType == codesign.APIKeyAuth || authType == codesign.AppleIDAuth {
+		if serviceConnection, err = devPortalClientFactory.CreateBitriseConnection(config.BuildURL, string(config.BuildAPIToken)); err != nil {
+			return codesign.Manager{}, err
+		}
+	}
+
+	appleAuthCredentials, err := codesign.SelectConnectionCredentials(authType, serviceConnection, logger)
+	if err != nil {
+		return codesign.Manager{}, err
+	}
+
+	opts := codesign.Opts{
+		AuthType:                   authType,
+		ShouldConsiderXcodeSigning: true,
+		TeamID:                     "",
```

Comment:
Why is this empty?

Reply:
There is no team ID input, so can not specify it.
Team ID would be used for selecting between teams (if the Apple ID is part of multiple ones), only foe Apple ID.
It is also used to filter certificates. It can work without it except if there is no team ID in the project and Apple ID with multiple memberships is used.